go.work

```diff
@@ -0,0 +1,725 @@
+go 1.18
+
+use (
+	.
+	./staging/src/k8s.io/api
+	./staging/src/k8s.io/apiextensions-apiserver
+	./staging/src/k8s.io/apimachinery
+	./staging/src/k8s.io/apiserver
+	./staging/src/k8s.io/cli-runtime
+	./staging/src/k8s.io/client-go
+	./staging/src/k8s.io/cloud-provider
+	./staging/src/k8s.io/cluster-bootstrap
+	./staging/src/k8s.io/code-generator
+	./staging/src/k8s.io/code-generator/examples
+	./staging/src/k8s.io/component-base
+	./staging/src/k8s.io/component-helpers
+	./staging/src/k8s.io/controller-manager
+	./staging/src/k8s.io/cri-api
+	./staging/src/k8s.io/csi-translation-lib
+	./staging/src/k8s.io/kube-aggregator
+	./staging/src/k8s.io/kube-controller-manager
+	./staging/src/k8s.io/kube-proxy
+	./staging/src/k8s.io/kube-scheduler
+	./staging/src/k8s.io/kubectl
+	./staging/src/k8s.io/kubelet
+	./staging/src/k8s.io/legacy-cloud-providers
+	./staging/src/k8s.io/metrics
+	./staging/src/k8s.io/mount-utils
+	./staging/src/k8s.io/pod-security-admission
+	./staging/src/k8s.io/sample-apiserver
+	./staging/src/k8s.io/sample-cli-plugin
+	./staging/src/k8s.io/sample-controller
+)
+
+replace bazil.org/fuse => bazil.org/fuse v0.0.0-20160811212531-371fbbdaa898
```

Comment:
Why do we need all these `replace` statements?

My understanding is that without them, Go will use the `replace` statements from the different `go.mod` files, so we shouldn't need to replicate them.

As an experiment, I ripped them all out. "make all" still worked, after resolving one problem: `staging/src/k8s.io/code-generator/examples` is not consistent with the other packages, probably because update-vendor.sh ignores it (?). We probably should fix that or, if that doesn't work, not include it in go.work file.

 However, after "make clean" the same command failed:
 ```
+++ [0608 10:39:54] Building go targets for linux/amd64
    ./staging/src/k8s.io/code-generator/cmd/deepcopy-gen (non-static)
make[1]: *** No rule to make target '.make/deps/by-path/./staging/src/k8s.io/code-generator/examples/HyphenGroup/apis/example/v1/_pkg', needed by 'staging/src/k8s.io/code-generator/examples/HyphenGroup/apis/example/v1/zz_generated.deepcopy.go'.  Stop.
make: *** [Makefile:536: generated_files] Error 2
```

So the replace statements are needed for something, it's just not entirely clear to me for what.

Reply:
It worked without any `replace` statements after re-adding `./staging/src/k8s.io/code-generator/examples` and fixing its `go.mod` file.

Reply:
See https://github.com/thockin/kubernetes/pull/12